### PR TITLE
[zk-sdk] Add batched range proof instructions

### PR DESCRIPTION
**Pull request by samkim-crypto**
#### Problem
The batched range proof instructions have yet to be added to the elgamal proof program.

#### Summary of Changes
Add the batched range proof instructions. I think all the changes should be pretty straightforward. I think the regular (non-batched) range proof instruction can be subsumed by the batched range proof instructions, so I don't think we need to add it.

Fixes #

### CODE REVIEW

File: zk-sdk/src/elgamal_program/proof_data/batched_range_proof/batched_range_proof_u128.rs [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+            vec![
+                &opening_1, &opening_2, &opening_3, &opening_4, &opening_5, &opening_6, &opening_7,
+                &opening_8,
+            ],
+        )
+        .unwrap();
+
+        assert!(proof_data.verify_proof().is_ok());
+
+        let amount_1 = 65536_u64; // not representable as an 8-bit number
```
Comment (joncinque) [inline zk-sdk/src/elgamal_program/proof_data/batched_range_proof/batched_range_proof_u128.rs]: micro-nit:
```suggestion
        let amount_1 = 65536_u64; // not representable as a 16-bit number
```

File: zk-sdk/src/elgamal_program/proof_data/batched_range_proof/batched_range_proof_u256.rs [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+            vec![
+                &opening_1, &opening_2, &opening_3, &opening_4, &opening_5, &opening_6, &opening_7,
+                &opening_8,
+            ],
+        )
+        .unwrap();
+
+        assert!(proof_data.verify_proof().is_ok());
+
+        let amount_1 = 4294967296_u64; // not representable as an 8-bit number
```
Comment (joncinque) [inline zk-sdk/src/elgamal_program/proof_data/batched_range_proof/batched_range_proof_u256.rs]: micro-nit:
```suggestion
        let amount_1 = 4294967296_u64; // not representable as a 32-bit number
```

File: zk-sdk/src/elgamal_program/proof_data/batched_range_proof/mod.rs [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+//! `VerifyBatchedRangeProof128`, and `VerifyBatchedRangeProof256`. The value `N` in
+//! `VerifyBatchedRangeProof{N}` specifies the sum of the bit-lengths that the proof is certifying
+//! for a sequence of commitments.
+//!
+//! For example to generate a batched range proof on a sequence of commitments `[C_1, C_2, C_3]` on
+//! a sequence of bit-lengths `[32, 32, 64]`, one must use `VerifyBatchedRangeProof128` as 128 is
+//! the sum of all bit-lengths.
+//!
+//! The maximum number of commitments is fixed at 8. Each bit-length in `[n_1, ..., n_N]` must be a
+//! power-of-two positive integer less than 128.
```
Comment (joncinque) [inline zk-sdk/src/elgamal_program/proof_data/batched_range_proof/mod.rs]: micro-nit: is this correct?
```suggestion
//! The maximum number of commitments is fixed at 8. Each bit-length in `[n_1, ..., n_N]` must be a
//! power-of-two positive integer less than or equal to 128.
```

**Review by joncinque — COMMENTED** [review]
just nits on the comments, looks great otherwise!

File: zk-sdk/src/elgamal_program/proof_data/batched_range_proof/mod.rs (hunk 3 above)
Comment (samkim-crypto) [reply zk-sdk/src/elgamal_program/proof_data/batched_range_proof/mod.rs]: Ah yes, that is correct. Thanks for the catch!

File: zk-sdk/src/elgamal_program/proof_data/batched_range_proof/batched_range_proof_u128.rs (hunk 1 above)
Comment (samkim-crypto) [reply zk-sdk/src/elgamal_program/proof_data/batched_range_proof/batched_range_proof_u128.rs]: Oops, sorry for these sloppy errors...

**Review by joncinque — APPROVED** [review]

**Pull request merged**